src/Workspaces/Core/Portable/SolutionCrawler/BackgroundAnalysisScope.cs

```diff
@@ -10,9 +10,10 @@ internal enum BackgroundAnalysisScope
         //       and that would break users who have saved a non-default scope for the option.
 
         /// <summary>
-        /// Analyzers are executed only for currently active document.
+        /// Analyzers are computed for visible documents
+        /// and open documents which had errors/warnings in the prior solution snapshot.
```

Comment:
this could use some more text.  explain the part about FilesWithPreviouslyReportedDiagnostics.  LIke why we want taht. 

Also if this is only for open files, then rename this to VisibleFilesAndOpenFilesWithPreviouslyReportedDiagnostics

Reply:
Done, also added more details to the doc comment